main.go

```
@@ -704,6 +706,28 @@ func (app *application) otherHandler(w http.ResponseWriter, r *http.Request) {
 </html>`)
 }
 
+func (app *application) sleepHandler(w http.ResponseWriter, r *http.Request) {
+	sleep := 30
+	if t, err := strconv.Atoi(r.URL.Query().Get("t")); err == nil {
+		sleep = t
```

Maybe we should return an error if strconv fails?

---

It's an option. Would you return a 422 HTTP status code if `t` is unset or can not be casted to `int`?

---

Yeah, or just a `400`, happy with either. Just something to let the user know if they use a non-int value when working with this endpoint.

---

See [d93899f](https://github.com/ankur22/testserver/pull/3/commits/d93899ff002ffef695e630d99ae27294a0e6b66b).